pkg/autodiscover/autodiscover_operators.go

```diff
@@ -63,32 +64,50 @@ func isIstioServiceMeshInstalled(allNs []string) bool {
 	return true
 }
 
-func findOperatorsByLabel(olmClient clientOlm.Interface, labels []labelObject, namespaces []configuration.Namespace) []*olmv1Alpha.ClusterServiceVersion {
-	csvs := []*olmv1Alpha.ClusterServiceVersion{}
+func findLabelSelector(labels []labelObject) string {
+	labelSelector := ""
+	if len(labels) > 0 {
+		labelSelector = labels[0].LabelKey + "=" + labels[0].LabelValue
+		for _, l := range labels[1:] {
```

Comment:
I believe that this would try to match pods matching all labels, instead of pods matching any labels